### PR TITLE
[FLINK-13691][Table]Remove deprecated query config

### PR DESCRIPTION
**Pull request by manbuyun**
## What is the purpose of the change

In Flink 1.9 the query config was marked deprecated and integrated into TableConfig. We should remove it from the API to reduce complexity and confusion

## Brief change log

- remove StreamQueryConfig
- remove BatchQueryConfig
- remove QueryConfig

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes)
  - The serializers: (yes)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by JingsongLi** [comment]
Hi @manbuyun , can you reply to @twalthr in the JIRA first?

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/PlannerConfig.java [hunk 1]
```diff
@@ -20,13 +20,14 @@
...
 
+import java.io.Serializable;
 import java.util.Optional;
 
 /**
  * The {@link PlannerConfig} holds parameters to configure the behavior of queries.
  */
 @PublicEvolving
-public interface PlannerConfig {
+public interface PlannerConfig extends Serializable {
```
Comment (JingsongLi) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/PlannerConfig.java]: Why do this?

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableConfig.java [hunk 2]
```diff
@@ -60,7 +61,7 @@
  * @see OptimizerConfigOptions
  */
 @PublicEvolving
-public class TableConfig {
+public class TableConfig implements Serializable {
```
Comment (JingsongLi) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableConfig.java]: Why need `Serializable`?

**Comment by manbuyun** [comment]
@flinkbot

**Comment by JingsongLi** [comment]
@flinkbot run travis

**Comment by manbuyun** [comment]
@flinkbot run travis

**Comment by manbuyun** [comment]
@flinkbot run azure

**Comment by JingsongLi** [comment]
@manbuyun can you check the failed tests? Like `ScalaShellITCase. testGroupedAggregationStreamTableAPIQuery `

**Comment by JingsongLi** [comment]
@flinkbot run azure

**Comment by JingsongLi** [comment]
Merging...

**Pull request merged**